From b7aefee65b219a08c8ddf45104f77fe84a2f4455 Mon Sep 17 00:00:00 2001
From: Christian Marangi <ansuelsmth@gmail.com>
Date: Tue, 8 Apr 2025 17:46:54 +0200
Subject: [PATCH 1/2] net: phy: mediatek: permit to compile test GE SOC PHY
 driver

When commit 462a3daad679 ("net: phy: mediatek: fix compile-test
dependencies") fixed the dependency, it should have also introduced
an or on COMPILE_TEST to permit this driver to be compile-tested even if
NVMEM_MTK_EFUSE wasn't selected. The driver makes use of NVMEM API that
are always compiled (return error) so the driver can actually be
compiled even without that config.

Fixes: 462a3daad679 ("net: phy: mediatek: fix compile-test dependencies")
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
---
 drivers/net/phy/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/phy/Kconfig b/drivers/net/phy/Kconfig
index 2a8ac5aed0f8..c80b4c5b7b66 100644
--- a/drivers/net/phy/Kconfig
+++ b/drivers/net/phy/Kconfig
@@ -16,7 +16,7 @@ config MEDIATEK_GE_PHY
 config MEDIATEK_GE_SOC_PHY
 	tristate "MediaTek SoC Ethernet PHYs"
 	depends on (ARM64 && ARCH_MEDIATEK) || COMPILE_TEST
-	depends on NVMEM_MTK_EFUSE
+	depends on NVMEM_MTK_EFUSE || COMPILE_TEST
 	help
 	  Supports MediaTek SoC built-in Gigabit Ethernet PHYs.
 
-- 
2.48.1

